src/Components/Components/src/ComponentBase.cs

```diff
@@ -298,9 +319,12 @@ Task IHandleEvent.HandleEventAsync(EventCallbackWorkItem callback, object arg)
 
         Task IHandleAfterRender.OnAfterRenderAsync()
         {
-            OnAfterRender();
+            var firstRender = !_hasCalledOnAfterRender;
+            _hasCalledOnAfterRender |= true;
```

Comment:
`_hasCalledOnAfterRender = true;` Why you being fancy?

Reply:
because I'm a fancy boy.